idx/bigtable/schema.go

```diff
@@ -0,0 +1,99 @@
+package bigtable
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"strings"
+
+	"cloud.google.com/go/bigtable"
+	"github.com/grafana/metrictank/idx"
+	"github.com/raintank/schema"
+	log "github.com/sirupsen/logrus"
+)
+
+func SchemaToRow(def *schema.MetricDefinition) map[string][]byte {
+	row := map[string][]byte{
+		"Id":         []byte(def.Id.String()),
+		"OrgId":      make([]byte, 8),
+		"Name":       []byte(def.Name),
+		"Interval":   make([]byte, 8),
+		"Unit":       []byte(def.Unit),
+		"Mtype":      []byte(def.Mtype),
+		"Tags":       []byte(strings.Join(def.Tags, ";")),
+		"LastUpdate": make([]byte, 8),
+		"Partition":  make([]byte, 8),
+	}
+	binary.PutVarint(row["OrgId"], int64(def.OrgId))
+	binary.PutVarint(row["Interval"], int64(def.Interval))
+	binary.PutVarint(row["LastUpdate"], def.LastUpdate)
+	binary.PutVarint(row["Partition"], int64(def.Partition))
+	return row
+}
+
+func RowToSchema(row bigtable.Row, def *schema.MetricDefinition) error {
```

Comment:
what's the purpose of requiring a pre-existing pointer to be passed in?

Reply:
Allow the caller to use a sync.Pool to reduce allocations.